task8/src/main/java/AdCollection/AdService.java

```diff
@@ -0,0 +1,158 @@
+
+                new Ad(2, "Репетиторы", "2021-01-01T11:00:00", "https://profi-bel.by/repetitor/",
+                "Степан", new ArrayList<String>(Arrays.asList("Образовательные услуги", "Прочие услуги")),
+                        "10", "2021-03-21T24:00:00",
+                "https://raw.githubusercontent.com/annapetrachkova/educational-practice-/UI/UI/assets/profi.jpg");
+
+                new Ad(3, "Маникюр и педикюр", "2020-12-12T11:00:00", "https://fokanail.by/",
+                "Алеся Хаткевич", new ArrayList<String>(Arrays.asList("Красота и здоровье")),
+                "30", "2021-03-31T24:00:00",
+                "https://raw.githubusercontent.com/annapetrachkova/educational-practice-/UI/UI/assets/nails.jpg");
+
+                new Ad(4, "Клининговая компания", "2020-12-01T11:00:00", "https://clean-up.by/",
+                "Евгения", new ArrayList<String>(Arrays.asList("Бытовые услуги", "Ремонт", "Прочие услуги")),
+                "15", "2021-04-09T24:00:00",
+                "https://raw.githubusercontent.com/annapetrachkova/educational-practice-/UI/UI/assets/clean.jpg");
+
+                new Ad(5, "Услуги фотошопа", "2020-11-14T11:00:00", "https://www.prettyphoto.ru/",
+                "ООО Боннети", new ArrayList<String>(Arrays.asList("Фото и видеосъемка", "Прочие услуги")),
+                "50", "2021-09-14T24:00:00",
+                "https://raw.githubusercontent.com/annapetrachkova/educational-practice-/UI/UI/assets/ps.jpg");
+
+                new Ad(10, "Макияж", "2021-08-01T11:00:00", "https://galinamakeup.by",
+                "Евгения", new ArrayList<String>(Arrays.asList("Красота и здоровье")),
+                "40", "2021-03-01T24:00:00",
+                "https://raw.githubusercontent.com/annapetrachkova/educational-practice-/UI/UI/assets/makeup.jpg'");
+
+    }
+
+    public List<Ad> getPage(int skip, int top, Map<String, String> filterConfig) {
+        List<Ad> filteredAds = new ArrayList<Ad>();
+        for (Map.Entry pair : filterConfig.entrySet()) {
+            if (pair.getKey().equals("vendor")) {
+                list.stream()
+                        .filter(adv -> adv.getVendor().toLowerCase().contains(pair.getValue().toString().toLowerCase()))
+                        .forEach(filteredAds::add);
+            } else if (pair.getKey().equals("creationDate")) {
+                list.stream()
+                        .filter(adv -> adv.getCreatedAt().equals(pair.getValue()))
+                        .forEach(filteredAds::add);
+            }
+        }
+        if (filterConfig.size() == 0) {
+            filteredAds = new ArrayList<Ad>(list);
+        }
+        if (top > filteredAds.size()) {
+            top = filteredAds.size();
+        }
+        if (skip >= filteredAds.size()) {
+            skip = 0;
+        }
+        if (skip + top > filteredAds.size()) {
+            return filteredAds.subList(skip, filteredAds.size());
+        } else {
```

Comment:
Then some `if` ends with `return`, the next `} else {` is redundant. Everything that goes after a "return" is by definition "else")